### ISSUE TITLE
Doesn't show up the nested form

### ISSUE THREAD

**Issue by adhamkurniawan** [nathanvda/cocoon#459]
I've following the instruction and everything seems work in model. But, when I open in browser it doesn't show the nested form in my form. I've created _customer_fields.html.erb inside my products folder view. And create this code in my _form.html.erb keep doesn't show up the form nested.
`
<%= form.simple_fields_for :customers do |customer| %>
    <div id="customers">
      <%= render 'customer_fields', :form => customer %>
    </div>
    <%= link_to_add_association 'add customer', form, :customers %>
<% end %>
`
Is this issue because the build: error above the readme?

**Comment by nathanvda** [comment]
The build error is because travis ci no longer supports ruby 2.0 (or works for ruby 2.0) and I have not yet fixed the build configuration. So no it is not because of the build configuration. 

Things to consider: 

- are there any nested items to show? if not, it is logical. If you want default nested items you should create one (some?) in your controller 
- check the [ERB examples](https://github.com/nathanvda/cocoon/wiki/ERB-examples) in the wiki (if you have trouble reading haml)

If you need more/better help, please give a little more information to work with. We want to see your models/associations, controller code  and view code (not complete but the minimal to reproduce and understand what you are doing and encountering) and either an error code or what you expect to see and what you miss.

**Comment by adhamkurniawan** [comment]
It's just one nested form inside product have a form customer. I did scaffold product and generate model customer as in instruction. Inside my product controller I added customers_attributes:
`
def product_params
  params.require(:product).permit(:name, :qty, :price, customers_attributes: [:id, :name, :phone])
end
`
Then I added this code inside _form.html.erb product:
`
<%= form.simple_fields_for :customers do |customer| %>
  <div id="customers">
    <%= render 'customer_fields', :form => customer %>
  </div>
  <%= link_to_add_association 'add customer', form, :customers %>
<% end %>
`
Next created separate _customer_fields.html.erb:
`
<div class="nested-fields">
  <div class="field">
    <%= form.label :name %>
    <%= form.text_field :name %>
  </div>

  <div class="field">
    <%= form.label :phone %>
    <%= form.text_field :phone %>
  </div>

  <%= link_to_remove_association "remove task", form %>
</div>
`
And in product model and customer:
Prouct.rb
`
has_many :customers, inverse_of: :product
accepts_nested_attributes_for :customers, reject_if: :all_blank, allow_destroy: true
`
Customer.rb
`
belongs_to :product
`

When I run rails s, it still not showing the nested form

**Comment by nathanvda** [comment]
Excuse me if this sounds overly detailed, but I am confused, because you mention running `rails s` (which seems obvious and not worth mentioning). So you are visiting a url, e.g. something like `http://localhost:3000/products/1/edit` ,  and you can then see the form, yes? 

When you see the form you do not see the nested form for a customer, because there is no customer? Or you do not see the form, but you know the product has a customer in the database? You do see the button to add a new customer?

**Comment by adhamkurniawan** [comment]
Sorry about that. I mean, when I try to create new product the nested form of customer doesn't show up in product form. It should showing the nested form actually in new form of product.
![screenshot from 2017-10-01 13-37-38](https://user-images.githubusercontent.com/13554803/31052402-c4b86a12-a6ad-11e7-9740-f082cc2317d2.png)

**Comment by nathanvda** [comment]
Ok, I think we see two errors, the first in your form. You should check the ERB examples as mentioned before, but you place the `link_to_add_association` inside the `simple_fields_for` loop, so it will not appear when no nested items are available, so the correct form would be: 

```
    <div id='customers'>
      <%= form.simple_fields_for :customers do |customer| %>
        <%= render 'customer_fields', :form => customer %>
      <% end %>
      <div class='links'>
        <%= link_to_add_association 'add task', f, :tasks %>
      </div>
    </div>
```

Secondly, the error suggests your `customers` data-model does not have a `product_id` field? (you will have to define this field inside a migration)

**Comment by adhamkurniawan** [comment]
After I replace `link_to_add_association` outside form tag, it shows unknown attribute product_id. My question is I want to add nested form customer inside product and the table product just need customer_id why I should add product_id too inside customers table? In product model we declare `has_many :customers, inverse_of: :product` and `accepts_nested_attributes_for :customers, reject_if: :all_blank, allow_destroy: true` which is we call cocoon functionality.
![screenshot from 2017-10-06 09-25-42](https://user-images.githubusercontent.com/13554803/31260468-586f6f58-aa78-11e7-8bd7-0e53b3639dac.png)

**Comment by nathanvda** [comment]
Obviously you can not place the `link_to_add_association` outside form tag, but I guess you mean (from the screenshot), outside the `simple_fields_for` loop.

You have not shown me the associations, but when your product has `have_many :customers` then the `customers` table needs to have a column called `product_id` to refer to it and to be able to write `belongs_to :product` in your code. Please check the rails association documentation http://guides.rubyonrails.org/association_basics.html

Simply put: if a product has `customer_id`, a product can only ever link to a single customer, if customers point to a product (have a `product_id` column), many customers can point to the same product, which is needed for a has-many relationship.
